File: front/src/js/Books/mutations.ts
import gql from 'graphql-tag'

export const BOOK_CREATE_MUTATION = gql`
    mutation BookCreateMutation($input: BookInputType!) {
        bookCreate(input: $input){
            book {
                id
                title
            }
        }
    }
`

export const BOOK_DELETE_MUTATION = gql`
    mutation BookDelete($id: ID!){
        bookDelete(id: $id){
            ok
        }
    }
`

File: back/gql/books/mutations.py
from graphene import Boolean, Field, ID, InputObjectType, Mutation, String, Date
from rest_framework import serializers

from books.models import Book
from .types import BookType


class BookSerializer(serializers.ModelSerializer):
    class Meta:
        model = Book
        fields = (
            'id',
            'title',
            'year'
        )

class BookInputType(InputObjectType):
    title = String()
    year = Date()


class BookCreate(Mutation):
    class Arguments:
        input = BookInputType(required=True)

    book = Field(BookType)

    @classmethod
    def mutate(cls, root, info, **data):
        serializer = BookSerializer(data=data.get('input'))
        serializer.is_valid(raise_exception=True)

        return BookCreate(book=serializer.save())


class BookDelete(Mutation):
    class Arguments:
        id = ID(required=True)

    ok = Boolean()

    @classmethod
    def mutate(cls, root, info, **data):
        book = Book.objects.get(id=data.get('id'))
        book.delete()

        return BookDelete(ok=True)